spec.html

```diff
@@ -38326,6 +38326,126 @@ <h1>`Promise.all` Resolve Element Functions</h1>
         </emu-clause>
       </emu-clause>
 
+      <emu-clause id="sec-promise.allsettled">
+        <h1>Promise.allSettled ( _iterable_ )</h1>
+        <p>The `allSettled` function returns a promise that is fulfilled with an array of promise state snapshots, but only after all the original promises have settled, i.e. become either fulfilled or rejected. It resolves all elements of the passed iterable to promises as it runs this algorithm.</p>
+        <emu-alg>
+          1. Let _C_ be the *this* value.
+          1. If Type(_C_) is not Object, throw a *TypeError* exception.
```

Comment:
This line can be removed, to match `Promise.race` and `Promise.all`. Doing so does not change observable semantics, since if `Type(C)` is not `Object`, then `NewPromiseCapability` in the next step throws the same exception anyhow.